lib/exla/shape.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Exla.Shape do
+  alias __MODULE__, as: Shape
+
+  @enforce_keys [:dims, :dtype, :ref]
+  defstruct [:dims, :dtype, :ref]
+
+  # TODO: convert dtype to integer representation to use in NIF
+  def make_shape(dtype, dims) when is_atom(dtype) and is_tuple(dims) do
```

Comment:
Should dims be a list or a tuple? Since it can have arbitrary length... maybe a list is better? What would be the biggest number of dimensions we would work with? 10 or 1000?

Reply:
For shapes I think a Tuple is better. In deep learning the highest number of dimensions I've seen is 5 for working with video data (batch size, frame, height, width, color channels).

It's not very common to go above that. I guess it's possible but most of the time you're working with 2, 3, or 4 dimensions.

I thought a list wouldn't necessarily be needed because you have to individually define the size of a dimension so I can't imagine anybody defining something with 10, 100, or 1000 dimensions